Makefile

```diff
@@ -94,7 +94,8 @@ endif
 
 REPO_BRANCH=$(shell git rev-parse --abbrev-ref HEAD | tr / _)
 REPO_SHA=$(shell git describe --match '$$^' --abbrev=8 --always --dirty)
-REPO_TAG=$(shell git describe --always --tags | grep -E '[0-9]*\.[0-9]*\.[0-9]*' || echo snapshot)
+# set this variable to the current tag only if we are building from a tag (annotated or not), otherwise set it to "snapshot", which means rootfs version will be constructed differently
+REPO_TAG=$(shell git describe --always --tags | grep -E '^[0-9]*\.[0-9]*\.[0-9]*$$' || echo snapshot)
```

Comment:
Can we modify the regex to `(^[0-9]+\.[0-9]+\.[0-9]+$$)`

> Reason: The first pattern (^[0-9]+\.[0-9]+\.[0-9]+$$) ensures that each part of the version (major, minor, patch) has at least one digit, which is a more accurate representation of semantic versioning. The second pattern (^[0-9]*\.[0-9]*\.[0-9]*$$) allows empty strings between dots, which is not typical in semantic versioning.

Reply:
thanks, that's definitely better!